spark/src/main/scala/org/apache/spark/sql/delta/commands/CreateDeltaTableCommand.scala

```diff
@@ -111,6 +112,13 @@ case class CreateDeltaTableCommand(
     } else if (mode == SaveMode.ErrorIfExists && tableExistsInCatalog) {
       throw DeltaErrors.tableAlreadyExists(table)
     }
+    val tableFeatures = TableFeatureProtocolUtils.
+      getSupportedFeaturesFromTableConfigs(table.properties)
+    if (!Utils.isTesting && (tableFeatures.contains(CatalogOwnedTableFeature) ||
```

Comment:
Add a comment to mention:
- This check should be relaxed once the catalog client (UCSingleCatalog) properly supports creating tables
- We bypass it in tests that can use the test client (InMemoryCommitCoordinator) to create tables.